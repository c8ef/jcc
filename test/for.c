int main() {
  int x;
  for (x = 1; x < 5; x += 1) {
  }
}
