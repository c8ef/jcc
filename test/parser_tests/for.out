FunctionDecl: main
  Args[0]
  CompoundStatement
    DeclStatement
      VarDecl: x
    ForStatement:
      ExprStatement
        BinaryExpr(=):
          DeclRefExpr: x
          IntergerLiteral: 1
      ExprStatement
        BinaryExpr(<):
          DeclRefExpr: x
          IntergerLiteral: 5
      BinaryExpr(+=):
        DeclRefExpr: x
        IntergerLiteral: 1
      CompoundStatement
